CONTRIBUTING.md

```diff
@@ -0,0 +1,81 @@
+# Contributing to Caliptra
+
+## CLA
+
+Per the [Charter](Caliptra_WG_Technical_Charter.md), contributors to Caliptra
+must have signed the CHIPS CLA.
+
+## Guidelines for code contributions
+
+* Changes must be made via pull request
+* [Branch protection rules](github_rules.md) ensure pull requests are reviewed
+  by project members
+* [Maintainers](MAINTAINERS.md) are expected to enforce `CODEOWNERS` rights
+  where necessary. Maintainers are also the sole owners of the repo's
+  configuration.
+* The branch protection rules include constraints to maintain a linear history
+  and avoid merges in pull requests.
+* Please keep a clean commit history, and no long series
+  of "fixup" patches. Use github's "Squash and merge" option if necessary.
+* New code files must include the Apache license 2.0 header. Commit veification
+  tools enforce this.
+* Fork the repos into your personal repo, and created pull requests from your
```

Comment:
This should be enforceable through Github.

Reply:
@korran can we prevent personal branches? not sure we can

Reply:
I don't think we can prevent this by construction, Steven

Reply:
Understood.

Reply:
We can strongly discourage this by adding a check workflow to the main branch that fails if the PR originates from a local branch.